exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/join/HashJoinBatch.java

```diff
@@ -1054,12 +1104,15 @@ public IterOutcome executeBuildPhase() throws SchemaChangeException {
       }
     }
 
+    int numPartitionsSpilled = 0;
```

Comment:
Please remove this variable. The PartitionStatSet already keeps track of the number of spilled partitions.

Reply:
Done; now using `buildPartitionStatSet.getNumSpilledPartitions()` to get the number of spilled partitions (just like the regular Hash Join memory calculator.